internal/signaling/stores/memory.go

```diff
@@ -191,7 +186,17 @@ func (m *Memory) Subscribe(ctx context.Context, topic string, callback func(cont
 			}
 			m.mutex.Unlock()
 
-			close(channel)
+			// Flush the channel.
+			// Since the channel isn't in m.topics[topic] anymore
+			// this shouldn't take more than a second.
+			timeout := time.After(time.Second)
+			for {
+				select {
+				case <-channel:
+				case <-timeout:
+					return
+				}
+			}
```

Comment:
Bit ugly but it should work slightly better than `close` and the `recover` in `Publish`.

Reply:
Why flush it at all?

Reply:
Otherwise a writer that got the channel already but didn't write on it yet will block on it forever.
That's why we only need to try reading for a short while since there is basically no time between getting the channel and writing to it in the writer.